examples/reinforcement_learning/ddpg_pendulum.py

```diff
@@ -0,0 +1,244 @@
+
+    Making tau close to 0 slows the pace of updates, and close to 1 might lead
+    to faster, but more volatile updates.
+    """
+    # Sort params by name
+    source_params = [param for _, param in sorted(source.namedparams())]
+    target_params = [param for _, param in sorted(target.namedparams())]
+    for s, t in zip(source_params, target_params):
+        t.data[:] += tau * (s.data - t.data)
+
+
+def main():
+
+    parser = argparse.ArgumentParser(description='Chainer example: DDPG')
+    parser.add_argument('--env', type=str, default='Pendulum-v0',
+                        help='Name of the OpenAI Gym environment to play')
+    parser.add_argument('--batch-size', '-b', type=int, default=64,
+                        help='Number of transitions in each mini-batch')
+    parser.add_argument('--episodes', '-e', type=int, default=1000,
+                        help='Number of episodes to run')
+    parser.add_argument('--gpu', '-g', type=int, default=-1,
+                        help='GPU ID (negative value indicates CPU)')
+    parser.add_argument('--out', '-o', default='ddpg_result',
+                        help='Directory to output the result')
+    parser.add_argument('--unit', '-u', type=int, default=100,
+                        help='Number of units')
+    parser.add_argument('--reward-scale', type=float, default=1e-3,
+                        help='Reward scale factor')
+    parser.add_argument('--replay-start-size', type=int, default=500,
+                        help=('Number of iterations after which replay is '
+                              'started'))
+    parser.add_argument('--tau', type=float, default=1e-2,
+                        help='Softness of soft target update (0, 1]')
+    parser.add_argument('--noise-scale', type=float, default=0.4,
+                        help='Scale of additive Gaussian noises')
+    parser.add_argument('--record', action='store_true', default=True,
+                        help='Record performance')
+    parser.add_argument('--no-record', action='store_false', dest='record')
+    args = parser.parse_args()
+
+    # Initialize an environment
+    env = gym.make(args.env)
+    assert isinstance(env.observation_space, gym.spaces.Box)
+    assert isinstance(env.action_space, gym.spaces.Box)
+    obs_size = env.observation_space.low.size
+    action_size = env.action_space.low.size
+    if args.record:
+        env = gym.wrappers.Monitor(env, args.out, force=True)
+    reward_threshold = env.spec.reward_threshold
+    if reward_threshold is not None:
+        print('{} defines "solving" as getting average reward of {} over 100 '
+              'consecutive trials.'.format(args.env, reward_threshold))
+    else:
+        print('{} is an unsolved environment, which means it does not have a '
+              'specified reward threshold at which it\'s considered '
+              'solved.'.format(args.env))
+
+    # Initialize variables
+    D = collections.deque(maxlen=10 ** 6)
```

Comment:
Similar with dqn example, it might be nice if there is one line comment that explains this variable. (D is a replay buffer)